test/Microsoft.Azure.WebJobs.Host.UnitTests/Executors/TriggeredFunctionExecutorTests.cs

```diff
@@ -53,11 +57,27 @@ public class TriggeredFunctionExecutorTests
                 InvokeHandler = invokeHandler
             };
 
+            Guid? expectedInvocationId = null;
+            if (passesInvocationId)
+            {
+                expectedInvocationId = Guid.NewGuid();
+                triggerData.Id = expectedInvocationId;
+            }
+
             var result = await triggerExecutor.TryExecuteAsync(triggerData, CancellationToken.None);
 
             Assert.True(result.Succeeded);
             Assert.True(customInvokerInvoked);
             Assert.True(innerInvokerInvoked);
```

Comment:
I think you should add a new targeted test for your scenario, rather than complicate this test. Your test could be simpler than this existing test I think.